mits/models.py

```diff
@@ -0,0 +1,179 @@
+from mits import *
+
+
+@Session.model_mixin
+class SessionMixin:
+    def logged_in_mits_team(self):
+        try:
+            return self.mits_team(cherrypy.session['mits_team_id'])
+        except:
+            raise HTTPRedirect('../mits_applications/team')
+
+    def mits_teams(self):
+        return (self.query(MITSTeam)
+                    .options(joinedload(MITSTeam.applicants),
+                             joinedload(MITSTeam.games),
+                             joinedload(MITSTeam.schedule),
+                             joinedload(MITSTeam.pictures))
+                    .order_by(MITSTeam.name))
+
+    def delete_mits_picture(self, picture):
+        self.delete(picture)
+        try:
+            os.remove(picture.filepath)
+        except:
+            log.error('unexpected error deleting MITS image {}', picture.filepath)
+        self.commit()
+
+
+class MITSTeam(MagModel):
+    name = Column(UnicodeText)
+    panel_interest = Column(Boolean, default=False)
+    want_to_sell = Column(Boolean, default=False)
+    address = Column(UnicodeText)
+    submitted = Column(UTCDateTime, nullable=True)
+
+    applied = Column(UTCDateTime, server_default=utcnow())
+    status = Column(Choice(c.MITS_APP_STATUS), default=c.PENDING, admin_only=True)
+
+    applicants = relationship('MITSApplicant', backref='team')
+    games = relationship('MITSGame', backref='team')
+    pictures = relationship('MITSPicture', backref='team')
+    schedule = relationship('MITSTimes', uselist=False, backref='team')
+
+    email_model_name = 'team'
+
+    @property
+    def accepted(self):
+        return self.status == c.ACCEPTED
+
+    @property
+    def email(self):
+        return [applicant.email for applicant in self.primary_contacts]
+
+    @property
+    def primary_contacts(self):
+        return [a for a in self.applicants if a.primary_contact]
+
+    @property
+    def salutation(self):
+        return ' and '.join(applicant.first_name for applicant in self.primary_contacts)
+
+    @property
+    def can_save(self):
+        return c.HAS_MITS_ADMIN_ACCESS or (self.is_new and c.BEFORE_MITS_SUBMISSION_DEADLINE or c.BEFORE_MITS_EDITING_DEADLINE)
+
+    @property
+    def completed_hotel_form(self):
+        return any(a.declined_hotel_space or a.requested_room_nights for a in self.applicants)
```

Comment:
Shouldn't this be all()..?

Reply:
Good question - I made it ``any`` rather than ``all`` deliberately because in theory someone can fill out their entire application, submit it, and then still go back in before the editing deadline has passed and add another person to their application.

In that case it didn't make intuitive sense to me to say that the application is suddenly no longer 100% complete, because we do indeed have everything we need to judge it.  Instead of downgrading their application to no longer being complete in this case it made more sense to me to handle that afterwards for already-accepted applications, e.g. with an automated email specifically for that purpose.

I've just added a docstring to this property explaining that, as well as adding the following comment to our ``automated_emails.py``:

```
# TODO: emails we still need to configure include but are not limited to:
# -> when teams have been accepted
# -> when teams have been declined
# -> when accepted teams have added people who have not given their hotel info
# -> final pre-event informational email
```